tests/ert_tests/ert3/config/test_stages_config.py

```diff
@@ -230,12 +235,12 @@ def test_stage_unknown_field(base_stage_config):
 
 def test_stage_input_immutable(base_stage_config):
     config = ert3.config.load_stages_config(base_stage_config)
-
+    input_record_name = base_stage_config[0]["input"][0]["record"]
     with pytest.raises(TypeError, match="does not support item assignment"):
-        config[0].input[0] = None
+        config[0].input[input_record_name] = None
```

Comment:
The proxy should support how it was, i.e. access via index, so why change the test?

Reply:
Old logic remaining from freezed-dict thinking